### ISSUE TITLE
qml cannot be validated - lxml issue?

### ISSUE THREAD

**Issue by filefolder** [obspy/obspy#2939]
Looks like this line in io/quakeml/core.py line 1895 is causing an exception in the current version of master.

`relaxng = RelaxNG(etree.parse(schema_location))`

RelaxNG imports fine and the file it's looking for, QuakeML-1.2.rng, is in the same place its always been and my lxml is version 4.7.1 which I think should be new enough.  Unfortunately I don't know enough about lxml to diagnose further. 

here's a quick test in case somehow it's just me

```
from obspy.clients.fdsn import Client
iris =Client('IRIS')
testcat = iris.get_events(starttime=obspy.UTCDateTime(2022,1,1),endtime=obspy.UTCDateTime(2022,1,1,0,5))
testcat.write('testcat.qml',format='QUAKEML',validate=True)

/home/seisop/.local/lib/python3.8/site-packages/obspy/io/quakeml/core.py:1898: UserWarning: Could not validate QuakeML - try using a newer lxml version
  warnings.warn(msg, UserWarning)
```

the file writes fine it just won't validate

**Comment by ThomasLecocq** [comment]
you have this warning an all python versions ? I remember lxml being a pain on py2.7, and somehow in 3.7 too

**Comment by filefolder** [comment]
haven't tried lower than 3.8 but at the moment

3.8.10 & lxml 4.7.1 = not working (master branch)
3.8.10 & lxml 4.7.1 = works (1.2.2) (!)
3.8.10 & lxml 4.6.3 = works (1.2.2)

different pc
3.8.10 & lxml 4.6.3 = works (1.2.2)

different pc
3.8.10 & lxml 4.5.2 = works (1.2.2)

different osx
3.9.7 & lxml 4.6.3 = works (OSX)  (1.2.2)

no idea why it would work or not work on the same lxml between 1.2.2 and now since the code looks the same?

**Comment by trichter** [comment]
Looking at

https://tests.obspy.org/119247/#21

```
Traceback (most recent call last):
File "[/home/runner/work/obspy/obspy/obspy/io/quakeml/core.py](https://github.com/obspy/obspy/blob/5f0068e2c2/obspy/io/quakeml/core.py#L1896)", line 1896, in _validate
relaxng = RelaxNG(etree.parse(schema_location))
File "src/lxml/etree.pyx", line 3536, in lxml.etree.parse
File "src/lxml/parser.pxi", line 1898, in lxml.etree._parseDocument
TypeError: cannot parse from 'PosixPath'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
File "[/home/runner/work/obspy/obspy/obspy/io/quakeml/tests/test_quakeml.py](https://github.com/obspy/obspy/blob/5f0068e2c2/obspy/io/quakeml/tests/test_quakeml.py#L1082)", line 1082, in test_avoid_empty_stub_elements
cat.write(memfile, format="QUAKEML", validate=True)
File "[/home/runner/work/obspy/obspy/obspy/core/event/catalog.py](https://github.com/obspy/obspy/blob/5f0068e2c2/obspy/core/event/catalog.py#L505)", line 505, in write
return write_format(self, filename, **kwargs)
File "[/home/runner/work/obspy/obspy/obspy/io/quakeml/core.py](https://github.com/obspy/obspy/blob/5f0068e2c2/obspy/io/quakeml/core.py#L1853)", line 1853, in _write_quakeml
if validate is True and not _validate(io.BytesIO(xml_doc)):
File "[/home/runner/work/obspy/obspy/obspy/io/quakeml/core.py](https://github.com/obspy/obspy/blob/5f0068e2c2/obspy/io/quakeml/core.py#L1899)", line 1899, in _validate
warnings.warn(msg, UserWarning)
UserWarning: Could not validate QuakeML - try using a newer lxml version
```

I think this error might have been introduced by #2751.

**Comment by trichter** [comment]
This is now fixed in #2981 

See also https://github.com/lxml/lxml/pull/337

**Comment by filefolder** [comment]
cheers, well done.

i hope these latest two PRs can be reviewed quickly
